File: goormlevel/175240/1. Stack/Stack.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {
	stack<int> s;
	int n = 0, k = 0;
	if (!(cin >> n >> k)) return 0;

	while (n--) {
		string cm;
		if (!(cin >> cm)) return 0;

		if (cm == "push") {
			int num;
			if (!(cin >> num)) return 0;

			if (s.size() < k) {
				s.push(num);
			}
			else {
				cout << "Overflow\n";
			}
		}
		else if (cm == "pop") {
			if (!s.empty()) {
				cout << s.top() << "\n";
				s.pop();
			}
			else {
				cout << "Underflow\n";
			}
		}
	}

	return 0;
}

File: goormlevel/175240/1. Stack/Stack.py
stack = []

n, k = map(int, input().split())

for _ in range(n):
	s = input().split()

	if s[0] == "push":
		num = int(s[1])

		if len(stack) < k:
			stack.append(num)
		else:
			print("Overflow")
	elif s[0] == "pop":
		if len(stack) != 0:
			print(stack.pop())
		else:
			print("Underflow")